#ROP Tutorial

This tutorial focuses on performing a comprehensive analysis of unmapped reads to profile repeats, circRNAs, gene fusions, trans-splicing events, recombined B/T-cell receptor sequences and microbial communities. This tutorial is a step-by-step description of the ROP (Read Origin Protocol) pipeline to explore the unmapped reads left from your study.

We assume you have a basic knowledge of sequence analysis and of Unix-based operating systems (although you should be able to run the pipeline on MacOS, some commands may require modification). If you have limited knowledge of UNIX, we encourage you to follow the [Unix Tutorial](https://github.com/smangul1/rop/wiki/UNIX-Tutorial). 

######[What is ROP?] (https://github.com/smangul1/rop/wiki/What-is-ROP%3F)
######[How ROP works?] (https://github.com/smangul1/rop/wiki/How-ROP-works%3F)
######[How to install ROP?] (https://github.com/smangul1/rop/wiki/How-to-install-ROP%3F)
######[Unix Tutorial] (https://github.com/smangul1/rop/wiki/UNIX-Tutorial)
######[Get started] (https://github.com/smangul1/rop/wiki/Get-started)
######[ROP analysis: one RNA-Seq sample] (https://github.com/smangul1/rop/wiki/ROP-analysis:--one-RNA-Seq-sample)
######[ROP output details] (https://github.com/smangul1/rop/wiki/ROP-output-details)
######[How to map reads and save unmapped reads?] (https://github.com/smangul1/rop/wiki/How-to-map-reads-and-save-unmapped-reads)
######[Source of every last read](https://github.com/smangul1/rop/wiki/Source-of-every-last-read)
######[Additional options](https://github.com/smangul1/rop/wiki/Additional-options)
######[How to calculate immune diversity?](https://github.com/smangul1/rop/wiki/How-to-calculate-immune-diversity%3F)



Please do not hesitate to contact us (smangul@ucla.edu) if you have any comments, suggestions, or clarification requests regarding the tutorial or if you would like to contribute to this resource.